services/director-v2/src/simcore_service_director_v2/modules/db/repositories/services_environments.py

```diff
@@ -46,11 +47,12 @@ async def get_vendor_secrets(
     def is_vendor_secret_identifier(cls, identifier: str) -> bool:
         return identifier.startswith(VENDOR_SECRET_PREFIX)
 
-    async def get_user_role(self, user_id: UserID):
+    async def get_user_role(self, user_id: UserID) -> UserRole:
         async with self.db_engine.acquire() as conn:
-            return UsersRepo().get_role(conn, user_id=user_id)
+            user_role = await UsersRepo().get_role(conn, user_id=user_id)
+            return UserRole(user_role)
 
     async def get_user_email(self, user_id: UserID) -> EmailStr:
         async with self.db_engine.acquire() as conn:
-            email = UsersRepo().get_email(conn, user_id=user_id)
+            email = await UsersRepo().get_email(conn, user_id=user_id)
```

Comment:
2 questions here:
1. why are you not using the users.py repository for this?
2. are you not using the linters? this should show up

Reply:
The issue is that `parse_obj_as` does not care what comes as input, even if it has a coroutine it will accept it.
In a similar fashion the above function was missing typing tighter with all its parent functions which was calling the. So no where along the chain this popped up.

Reply:
I have added tests to try and pick up these issues in the future

Reply:
still I don't understand why you duplicate that function that already exists in users.py??

Reply:
I've moved it to the correct place now